python/delta_sharing/tests/test_delta_sharing.py

```diff
@@ -280,6 +282,12 @@ def list_all_tables(
             pd.DataFrame({"c1": ["foo bar"], "c2": ["foo bar"],}),
             id="Azure Data Lake Storage Gen2",
         ),
+        pytest.param(
```

Comment:
@ueshin This test currently fails with below error. It looks like fsspec is not handling GCS file correctly. I'm not quite familiar with this part of the code. Do you happen to come across this issue before?

```
self = <pyarrow.fs.FSSpecHandler object at 0x126890070>, path = 'https://storage.googleapis.com/delta-sharing-dev/delta-sharing-test/table1/c2%3Dfoo%20bar/part-00000-3287e08f-94ef-44...3a8be15a0bd00c41b863a16e8f117ab911f99f10b09eeb8ec10b499f7d58b64bd2cbde845b3ab8604f4e1d83e0331d6ec44336cb45f6529538d90d'

    def open_input_file(self, path):
        from pyarrow import PythonFile
    
        if not self.fs.isfile(path):
>           raise FileNotFoundError(path)
E           FileNotFoundError: https://storage.googleapis.com/delta-sharing-dev/delta-sharing-test/table1/c2%3Dfoo%20bar/part-00000-3287e08f-94ef-44fe-9837-4d8eee39363f.c000.snappy.parquet?X-Goog-Algorithm=GOOG4-RSA-SHA256&X-Goog-Credential=delta-sharing-dev-svc%40databricks-dev.iam.gserviceaccount.com%2F20220107%2Fauto%2Fstorage%2Fgoog4_request&X-Goog-Date=20220107T234922Z&X-Goog-Expires=3600&X-Goog-SignedHeaders=host&X-Goog-Signature=0f78839681431f0996043fe7a6698c56ea2e3a12f6597f6fab572bb8a613eb37e33b26c393ad1802f1df2a1cabaaed4eb5486bb9f242e3bfa5c860c3a0b0f0ee542112c7ec3f311bed542a81577da6db230b7abb921dcb21df9eebb3a14bd14118f285c348bfeef8cccece063b45f6fceed8fc5f4ece4ed104639cedf7a0d313381c21968c6e6ad67a40700196fadddc38f6b278d621f740f4aa1eb9159f9e0b9df95afec1be601288171bd937fe5e11943c0cf0c134c25268d10900e6c078767e316699803a8be15a0bd00c41b863a16e8f117ab911f99f10b09eeb8ec10b499f7d58b64bd2cbde845b3ab8604f4e1d83e0331d6ec44336cb45f6529538d90d
```

Reply:
I dig the code and found a workaround for this file:

```
diff --git a/python/delta_sharing/reader.py b/python/delta_sharing/reader.py
index 6ab26fe..393b140 100644
--- a/python/delta_sharing/reader.py
+++ b/python/delta_sharing/reader.py
@@ -25,6 +25,12 @@ from delta_sharing.converter import to_converters, get_empty_table
 from delta_sharing.protocol import AddFile, Table
 from delta_sharing.rest_client import DataSharingRestClient
 
+try:
+    from yarl import URL
+    from yarl._quoting import _Quoter
+    URL._PATH_REQUOTER = _Quoter(safe="@:", protected="/+=")
+except:
+    pass
 
 class DeltaSharingReader:
     def __init__(
```
This means we will require yarl 1.6.0 (`_PATH_REQUOTER` is not in old versions).

The root cause is our client receives the raw url to send the requests. But when we pass it to fsspec, it will forward the url to aiohttp. There is a workaroud in https://github.com/aio-libs/aiohttp/issues/6413#issuecomment-993393969 to specify our url is an encoded url but we cannot change fsspec code.

The issue will cause we sent an incorrect url
```
https://storage.googleapis.com/delta-sharing-dev/delta-sharing-test/table1/c2=foo%20bar/part-00000-3287e08f-94ef-44fe-9837-4d8eee39363f.c000.snappy.parquet?...
```
rather than
```
https://storage.googleapis.com/delta-sharing-dev/delta-sharing-test/table1/c2%3Dfoo%20bar/part-00000-3287e08f-94ef-44fe-9837-4d8eee39363f.c000.snappy.parquet?...
```
(The difference of two urls is `c2=foo%20bar` and `c2%3Dfoo%20bar`)

This issue doesn't happen in AWS and Azure because they accept both urls, but GCS. will reject `c2=foo%20bar`.

The workaround I posted above is just making yarl.URL not change `%3D` to `=`.

`FileNotFoundError` is caused by https://github.com/fsspec/filesystem_spec/issues/873 . The real error dropped by pyarrow is
```
aiohttp.client_exceptions.ClientResponseError: 403, message='Forbidden', url=URL('https://storage.googleapis.com/delta-sharing-dev/delta-sharing-test/table1/c2=foo%20bar/part-...')
```